exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/join/TestHashJoinAdvanced.java

```diff
@@ -160,4 +166,75 @@ public void testJoinWithMapAndDotField() throws Exception {
       .baselineValues("1", "2", "1", null, "a")
       .go();
   }
+
+  private void buildFile(String fileName, String[] data, File testDir) throws IOException {
+    try(PrintWriter out = new PrintWriter(new FileWriter(new File(testDir, fileName)))) {
+      for (String line : data) {
+        out.println(line);
+      }
+    }
+  }
+
+  @Test
+  public void testHashLeftJoinWithEmptyTable() throws Exception {
+    ClusterFixtureBuilder builder = ClusterFixture.builder(dirTestWatcher);
+    File testDir = null;
+    try {
+      ClusterFixture cluster = builder.build();
+      ClientFixture client = cluster.clientFixture();
```

Comment:
Please see ExampleTest for how the cluster and client should be created in a try-with-resources block to ensure that things are closed.

Also, each cluster is started with the same parameters. The use of `ClusterTest` will allow you to start the cluster once, then reuse it for multiple test; which will speed up test cycle times.